### DISCUSSION TITLE
Save QuickAccessToolbar when closing

### DISCUSSION THREAD

**Discussion by Rickyd56** [fluentribbon/Fluent.Ribbon#1212] [Q&A] [answered]
I was wondering if you have an example on how to save the QAT Toolbar settings on window closing so they can be restore at the next startup?

I seen the discussion "QuiclAccessToolbar not save items. #969", but it only saves items that were changed during the current interaction.  I am hoping to save all of the toolbar settings even if they were not changed so they can be restored next time the app is opened..

Thank you in advance for any help.

Rick

**Comment by Rickyd56** [comment]
If there is a way to check the ischecked and isVisable states of a QAT Button, that could solve my challenge.  I would save the state in an xml file during the WindowClosing event and use it to set the checked state when the app is initialized.

**Comment by Rickyd56** [comment]
I did find a way to determine when the QAT Menuitem is check or unchecked, but still don't know how to set this value during initialization.
I am new to commands and bindings. 


 private void QATBrowser_IsChecked(object sender, RoutedEventArgs e)
 {
     Settings.Default.QAT_Browser_Checked = true;
 }

 private void QATBrowser_UnChecked(object sender, RoutedEventArgs e)
 {
     Settings.Default.QAT_Browser_Checked = false;
 }

**Comment by batzen** [comment, marked as answer]
Can't you just iterate over your QAT items and save/restore their state?
